darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,691 @@
+                        transformation["function"] in builtin_keys_list
+                        and set(
+                            itertools.chain(
+                                *self.BUILTIN_TRANSFORMS[transformation["function"]][2]
+                            )
+                        ).intersection(set(transformation.keys()))
+                        == set(
+                            itertools.chain(
+                                *self.BUILTIN_TRANSFORMS[transformation["function"]][2]
+                            )
+                        ),
+                        f"`window_transformation` dictionary at index {idx} must contain at least the following keys "
+                        f"{list(itertools.chain(*self.BUILTIN_TRANSFORMS[transformation['function']][2]))} "
+                        f"for built-in {self.BUILTIN_TRANSFORMS[transformation['function']][0].__name__}"
+                        f".{transformation['function']}"
+                        f".",
+                    )
+
+                if "window" in transformation:
+                    raise_if_not(
+                        transformation["window"] is not None,
+                        f"`window_transformation` dictionary at index {idx} must contain a non-empty 'window' key.",
+                    )
+
+                    if isinstance(transformation["window"], int):
+                        raise_if_not(
+                            transformation["window"] > 0,
+                            f"`window_transformation` at index {idx} must contain a positive integer for the 'window'.",
+                        )
+                        window_transformations[idx]["window"] = [
+                            transformation["window"]
+                        ]
+                    elif isinstance(transformation["window"], list):
+                        raise_if_not(
+                            len(transformation["window"]) > 0,
+                            f"`window_transformation` at index {idx} must contain a non-empty list for the 'window'. ",
+                        )
+
+                        for idws, window in enumerate(transformation["window"]):
+                            raise_if_not(
+                                isinstance(window, int) and window > 0,
+                                f"`window_transformation` at index {idx} must contain only positive integers for the "
+                                f"'window'. Found {window} at index {idws}.",
+                            )
+                    else:
+                        raise_log(
+                            ValueError(
+                                f"`window_transformation` at index {idx} must contain a positive integer or a list "
+                                f"of positive integers for the 'window'. "
+                            ),
+                            logger,
+                        )
+
+                if "step" in transformation:
```

Comment:
update : I kept 'step' as a key for this argument, because it is being automatically filtered along other potentially user provided arguments that I cannot tell in advance (more flexibility, but user should know a bit more if they want to configure more things). This key is an advanced usage, ideally the user should know what they are doing with it, if not, we set it to the safe value of 1 and they should not worry about it.